example/lib/pages/home.dart

```diff
@@ -58,9 +58,14 @@ class HomePage extends StatelessWidget {
                   zoom: 5,
                 ),
                 nonRotatedChildren: [
-                  AttributionWidget.defaultWidget(
-                    source: 'OpenStreetMap contributors',
-                    onSourceTapped: () {},
+                  //AttributionLayer.defaultWidget(
+                  //  source: '© OpenStreetMap contributors',
+                  //  onSourceTapped: () {},
+                  //),
+                  AttributionLayer(
+                    attributions: {
+                      const Text('© OpenStreetMap contributors'): () {}
```

Comment:
I think it would be worth refactoring this into an `AttributionSource`. It can have a detailed constructor like this with an onTapped action, but I would propose a convenience method as well to construct one with just a source name and a URL to launch on tap. This is (IMO) what 99.99% of them should be anyways. It would also save you from having to hunt around for the copyright glyph and would save a TON of boilerplate. Here's an example:

```dart
class AttributionSource {
  final String name;
  final void Function()? onSourceTapped;

  // TODO: Convenience constructor that accepts a URL rather than a function
}
```

The default onSourceTapped with said helper would look something like this:

```dart
() async {
  // Depends on package url_launcher
  if (!await launchUrl(uri)) {
    // Someone who actually understands dart logging correct this; if flutter_map doesn't already have a logger, maybe it should create one.
    log('Could not launch url', level: Level.WARNING.value);
  }
}
```

Reply:
Valid. I don't have a really strong opinion about this, but this is what like 99% of people probably want so it seems logical enough to add.

Does Dart have a package-level feature construct which conditionally includes code and deps based on feature flags (ex: cargo features in Rust)? You could solve any objection to the added weight by adding an opt-out this way.

Reply:
Unfortunately not, a package has to be referenced in the `pubspec.yaml`, even if it is not used in the project in the end, the code might not be included when building an application with tree shaking but the package dependency will remain.

Reply:
We can't depend on url_launcher because it requires extra setup for platform specifics: I would prefer to avoid this confusion.